packages/components/src/internal/components/editable/models.ts

```diff
@@ -554,54 +554,84 @@ export class EditorModel
 
     static getEditorDataFromQueryValueMap(valueMap: any): List<any> | any {
         // Editor expects to get either a single value or an array of an object with fields displayValue and value
-        if (valueMap && List.isList(valueMap)) {
-            return valueMap.map(val => {
-                // If immutable convert to normal JS
-                if (Iterable.isIterable(val)) {
-                    return { displayValue: val.get('displayValue'), value: val.get('value') };
-                } else return val;
-            });
-        } else if (
-            valueMap &&
-            valueMap.has('value') &&
-            valueMap.get('value') !== null &&
-            valueMap.get('value') !== undefined
-        ) {
-            return valueMap.has('formattedValue')
-                ? List<any>([{ displayValue: valueMap.get('formattedValue'), value: valueMap.get('value') }])
-                : valueMap.has('displayValue')
-                ? List<any>([{ displayValue: valueMap.get('displayValue'), value: valueMap.get('value') }])
-                : valueMap.get('value');
-        } else return undefined;
+        if (List.isList(valueMap)) {
+            return (valueMap as List<any>)
+                .map(val => {
+                    // If immutable convert to normal JS
+                    if (Iterable.isIterable(val)) {
+                        return { displayValue: val.get('displayValue'), value: val.get('value') };
+                    }
+                    return val;
+                })
+                .toList();
+        }
+
+        if (valueMap?.has('value') && valueMap.get('value') !== null && valueMap.get('value') !== undefined) {
+            if (valueMap.has('formattedValue')) {
+                return List([{ displayValue: valueMap.get('formattedValue'), value: valueMap.get('value') }]);
+            }
+            if (valueMap.has('displayValue')) {
+                return List([{ displayValue: valueMap.get('displayValue'), value: valueMap.get('value') }]);
+            }
+            return valueMap.get('value');
+        }
+
+        return undefined;
     }
 
     static convertQueryDataToEditorData(
         data: Map<string, any>,
         updates?: Map<any, any>,
         idsNotToUpdate?: number[],
-        fieldsNotToUpdate?: string[]
+        fieldsNotToUpdate?: string[],
+        encode = false // TODO: Defaulting to "false" to see outcomes in TC
```

Comment:
I'm assuming this is going to change before merging? Or at least remove the TODO if it's not problematic?

Reply:
Yes, removed.